### PR TITLE
feat: add experimental browser.debugInfo

### PR DESCRIPTION
**Pull request by OrKoN**
This is an experimental method that provides debugging information to help debug issues with Puppeteer.
Currently, it only provides the stacktraces for pending protocol calls.

Note: the test failure is unrelated and tracked in https://github.com/puppeteer/puppeteer/issues/11749 It's revealed by the change in sharding caused by introduction of a new spec.

### CODE REVIEW

File: packages/puppeteer-core/src/common/CallbackRegistry.ts [hunk 1]
```diff
@@ -94,6 +94,19 @@ export class CallbackRegistry {
     }
     this.#callbacks.clear();
   }
+
+  /**
+   * @internal
+   */
+  getPendingProtocolErrors(): Error[] {
+    const result: Error[] = [];
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/CallbackRegistry.ts:102]: Nit: Simplify to a map?

File: packages/puppeteer-core/src/api/Browser.ts [hunk 2]
```diff
@@ -431,4 +439,16 @@ export abstract class Browser extends EventEmitter<BrowserEvents> {
    * @internal
    */
   abstract get protocol(): ProtocolType;
+
+  /**
+   * Get debug information from Puppeteer.
+   *
+   * @remarks
+   *
+   * Currently, includes pending CDP calls. In the future, we might add more info.
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/api/Browser.ts]: From my understanding of the change this is not CDP but protocol here.

**Review by Lightning00Blade — APPROVED** [review]
LGTM + text change

File: test/TestExpectations.json [hunk 3]
```diff
@@ -3366,7 +3372,7 @@
     "testIdPattern": "[target.spec] Target Browser.waitForTarget should wait for a target",
     "platforms": ["darwin", "linux", "win32"],
     "parameters": ["chrome", "webDriverBiDi"],
-    "expectations": ["PASS"]
+    "expectations": ["FAIL", "PASS"]
```
Comment (jrandolf) [inline test/TestExpectations.json:3375]: Why is this failing?

**Review by jrandolf-google — APPROVED** [review]
lgtm % comments

File: test/TestExpectations.json (hunk 3 above)
Comment (OrKoN) [reply test/TestExpectations.json:3375]: See https://github.com/puppeteer/puppeteer/issues/11749 and chat.

**Comment by Lightning00Blade** [comment]
We miss adding this to the website, can you file an issue or PR for that?

**Pull request merged**